src/segment_reader.rs

```diff
@@ -342,6 +345,142 @@ impl AsyncSegmentReaderImpl {
     }
 }
 
+/// A wrapper around the AsyncSegmentReader.
+///
+/// It maintains a buffer and prefetches data to fill that buffer in the background so that
+/// every read call can read from the local buffer instead of waiting for server response.
+pub(crate) struct AsyncSegmentReaderWrapper {
+    buffer: Vec<u8>,
+    reader: Arc<Box<dyn AsyncSegmentReader>>,
+    offset: i64,
+    end_of_segment: bool,
+    is_truncated: bool,
+    outstanding: Outstanding,
+    handle: Handle,
+}
+
+impl AsyncSegmentReaderWrapper {
+    const DEFAULT_BUFFER_SIZE: usize = 1024 * 1024;
```

Comment:
Allow these buffer parameters to be specified by the user at runtime.

Reply:
got it